### ISSUE TITLE
Missing something basic.

### ISSUE THREAD

**Issue by alto777** [VCVRack/Rack#882]
I had a crashing plugin: wild pointers. I had to add to my step()

if (!runnable) reset();

where bool runnable is false until reset() (or onCreate()) is called. runnable false skips around the code in step() that was wildly pointing.

It seemed that neither reset() nor onCreate() was getting called automatically when I instantiated my plugin - not just a matter of not in time, but  _not at all_ as I concluded from a few lights set distinctively in reset() and onCreate() (printf for Rack). A manual Cmd-I runs reset() and would get things moving. Would something keep onCreate() from running, or do I misunderstand its role?

It works but it looks like 1) a kludge and 2) I am failing to understand something basic.

This was Rack 0.6.0, building with the Rack SDK.

**Comment by dllmusic** [comment]
It’s onReset() now

**Comment by AndrewBelt** [comment]
`reset()` is deprecated. Use `onReset()` instead. https://github.com/VCVRack/Rack/blob/v0.6/include/engine.hpp#L92

>Called when user clicks Initialize in the module context menu

https://github.com/VCVRack/Rack/blob/v0.6/include/engine.hpp#L76

You're falsing assuming it's called on initialization.
To call it on startup, call it from your module's constructor.

**Comment by alto777** [comment]
Thank you for trying to help! I tried some plausible variations and also found #690 to be informative. Unfortunately I was not successful variously. Some things wouldn't build and some built things did not do what I expected, so I replaced my cschol-style "one-shot" so I could get on w/ it.

I promise myself to learn enough to make sense of this. For now, I have achieved a level of incompetence which is allowing me to build amazing (!) stuff, you'll see one day I hope.

When is onCreate() called?

**Comment by ndrspcfd** [comment]
@alto777 did you try calling your onCreate() or onReset() from inside your module's constructor as Andrew advised?

**Comment by AndrewBelt** [comment]
Everything you want to know is in the documentation.

**Comment by alto777** [comment]
Thank you for your patience. I did try calling my onCreate() or onReset, as I said it either did not compile or if it did, did not do what I expected. Don't worry about that, **I'm in way over my head** so I probably did not manage to get something right.

So rather than another fender-bender left turn across traffic, I've settled on three right turns. Something more important is that my C++ class is quite crude and doesn't do enough checking, which would have saved me. I should protect myself and any who may ever see and use it. I am saved all the time by those who plan on my errors.

I see that onCreate() is
	/** Called when module is created by the Add Module popup, cloning, or when loading a patch or autosave */

_Curiously_, if I clone my stalled plugin, the clone fires up running, like onCreate was called. If I add my plugin from the pop-up, it comes in stalled. Again, no worries, as I have solved the alligator problem and returned to swamp-draining.
